### PR TITLE
fix(sdk): fix incorrect doc

### LINKED ISSUE #1474: [Bug]: SDK Trace propagator documentation is hidden.
### What happened?

Both `BaggagePropagator` and `TraceContextPropagator` contain some very useful module level documentation that is currently not shown because they`re private:

https://github.com/open-telemetry/opentelemetry-rust/blob/27d338dd954ef7681242ead47a8a61c964dcc428/opentelemetry-sdk/src/propagation/mod.rs#L1-L6

https://github.com/open-telemetry/opentelemetry-rust/blob/27d338dd954ef7681242ead47a8a61c964dcc428/opentelemetry-sdk/src/propagation/baggage.rs#L1-L42

https://github.com/open-telemetry/opentelemetry-rust/blob/27d338dd954ef7681242ead47a8a61c964dcc428/opentelemetry-sdk/src/propagation/trace_context.rs#L1-L19

Perhaps add the documentation to the respective structs?

### API Version

0.21

### SDK Version

0.21

### PR DESCRIPTION
**Pull request by TommyCpp**
Fixes #1474, #1494
Design discussion issue (if applicable) #

## Changes

- fix incorrect docs

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/propagation/baggage.rs [hunk 1]
```diff
@@ -1,45 +1,3 @@
-//! # OpenTelemetry Baggage API
```
Comment (TommyCpp) [inline opentelemetry-sdk/src/propagation/baggage.rs:1]: Removed this as it's duplicate with the `BaggatePropagator` doc

Comment (cijothomas) [reply opentelemetry-sdk/src/propagation/baggage.rs:1]: Thanks. 

Separately, https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/baggage.rs#L393-L399 has some extra things like `BaggageMetadata` that are not in spec.. Need to review them to ensure its needed.

Comment (cijothomas) [reply opentelemetry-sdk/src/propagation/baggage.rs:1]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#propagators-distribution 
Also, the propagators should be part of the API itself.. 

Not a blocker for this PR, will track separately.

File: opentelemetry-sdk/src/propagation/trace_context.rs [hunk 2]
```diff
@@ -33,8 +17,25 @@ const TRACESTATE_HEADER: &str = "tracestate";
 static TRACE_CONTEXT_HEADER_FIELDS: Lazy<[String; 2]> =
     Lazy::new(|| [TRACEPARENT_HEADER.to_owned(), TRACESTATE_HEADER.to_owned()]);
 
-/// Propagates `SpanContext`s in [W3C TraceContext] format.
+/// Propagates `SpanContext`s in [W3C TraceContext] format under `traceparent` header.
```
Comment (cijothomas) [inline opentelemetry-sdk/src/propagation/trace_context.rs:20]: its best to simply link to W3CTraceContext spec here?
Also, `traceparent` + `tracestate` both together makes the spec. This doc comment seems to indicate that `traceparent` alone is present..

**Review by cijothomas — APPROVED** [review]
some nits, not blockers

**Review by lalitb — APPROVED** [review]

File: opentelemetry-sdk/src/propagation/baggage.rs (hunk 1 above)
Comment (hdost) [reply opentelemetry-sdk/src/propagation/baggage.rs:1]: Is there a ticket to track?

Comment (cijothomas) [reply opentelemetry-sdk/src/propagation/baggage.rs:1]: No, haven;t create an issue to track yet.

File: opentelemetry-sdk/src/propagation/trace_context.rs (hunk 2 above)
Comment (TommyCpp) [reply opentelemetry-sdk/src/propagation/trace_context.rs:20]: > its best to simply link to W3CTraceContext spec here?

It should be linked already. The link is set on L39

File: opentelemetry-sdk/src/propagation/baggage.rs (hunk 1 above)
Comment (TommyCpp) [reply opentelemetry-sdk/src/propagation/baggage.rs:1]: > Also can't we reference it here without needing to duplicate?

We don' really expose this module(we export the `BaggagePropagator` directly) so users won't see it.

**Pull request merged**